tests/phpunit/integration/Core/User_Surveys/REST_User_Surveys_ControllerTest.php

```diff
@@ -103,6 +117,145 @@ public function test_register_routes() {
 		$this->assertEquals( \WP_REST_Server::READABLE, $args[0]['methods'] );
 		$this->assertTrue( is_callable( $args[0]['callback'] ) );
 		$this->assertTrue( is_callable( $args[0]['permission_callback'] ) );
+
+		$args = $routes['survey']->get_args();
+		$this->assertEquals( 'core/user/data/survey', $routes['survey']->get_uri() );
+		$this->assertEquals( \WP_REST_Server::READABLE, $args[0]['methods'] );
+		$this->assertTrue( is_callable( $args[0]['callback'] ) );
+		$this->assertTrue( is_callable( $args[0]['permission_callback'] ) );
+	}
+
+	protected function send_request( $endpoint, $data, $response ) {
+		remove_all_filters( 'googlesitekit_rest_routes' );
+		$this->controller->register();
+		$this->register_rest_routes();
+
+		$this->fake_proxy_site_connection();
+		$this->subscribe_to_wp_http_requests(
+			function() {},
+			array(
+				'response' => array( 'code' => 200 ),
+				'headers'  => array(),
+				'body'     => wp_json_encode( $response ),
+			)
+		);
+
+		$request = new WP_REST_Request( 'POST', '/' . REST_Routes::REST_ROOT . '/core/user/data/' . $endpoint );
+		$request->set_body_params(
+			array(
+				'data' => $data,
+			)
+		);
+
+		return rest_get_server()->dispatch( $request );
+	}
+
+	public function test_survey_trigger() {
+		$survey = array(
+			'survey_id' => 'test_survey',
+			'session'   => array(
+				'session_id'    => 'test_session_id',
+				'session_token' => 'test_session_token',
+			),
+			'payload'   => array(
+				'questions' => array(),
+			),
+		);
+
+		$response = $this->send_request(
+			'survey-trigger',
+			array( 'triggerID' => 'test_trigger' ),
+			$survey
+		);
+
+		$this->assertEqualSets( $survey, $response->get_data() );
+		$this->assertEqualSets( $survey, $this->queue->front() );
+	}
+
+	public function test_survey_event_survey_shown() {
+		$this->send_request(
+			'survey-event',
+			array(
+				'session' => array(
+					'session_id'    => 'test_session_id',
+					'session_token' => 'test_session_token',
+				),
+				'event'   => array(
+					'survey_shown' => array(),
+				),
+			),
+			array()
+		);
+
+		$this->assertEquals(
+			array( '__global' ),
+			$this->timeouts->get_survey_timeouts()
+		);
+	}
+
+	public function data_event_types() {
```

Comment:
How about something a little more descriptive as to which kind of event types? Maybe completion/interation types?

Reply:
Yep, updated to be `data_event_completion_types`.